nova/compute.sls

```diff
@@ -143,8 +143,21 @@ Add_compute_to_availability_zone_{{ compute.availability_zone }}:
   - name: "nova {{ identity_params }} aggregate-add-host {{ compute.availability_zone }} {{ pillar.linux.system.name }}"
   - unless: "nova {{ identity_params }} service-list | grep {{ compute.availability_zone }} | grep {{ pillar.linux.system.name }}"
 
+{%- if compute.aggregates != None %}
+
+{%- for aggregate in compute.aggregates %}
+Add_compute_to_aggregate_{{ aggregate }}:
```

Comment:
I'm missing part that enforces just the aggregate zone and part that is responsible for associating these existing zones to individual hypervisors.
to cope with the condifitions simon is referefing:
- run nova.client on ctls [enforces the zones and aggregates]
- run nova.compute [enforces the service]
- run nova.client on cmps [enforsces the zone/aggreg membership]